debian/postinst

```diff
@@ -59,9 +65,23 @@ upgrade_to_status_cache() {
 
 case "$1" in
     configure)
+      PREVIOUS_PKG_VER=$2
+      # Special case: legacy precise creds allowed for trusty esm
+      # do-release-upgrade substitutes s/precise/trusty/ in all apt sources.
+      # So all we need to do is rename the precise sources file to trusty.
+      # https://github.com/CanonicalLtd/ubuntu-advantage-client/issues/693
+      case $PREVIOUS_PKG_VER in
+          *12.04*) # upgraded from precise
```

Comment:
There is only one release of `ubuntu-advantage-tools` in precise and its version is `1`: https://launchpad.net/ubuntu/precise/+source/ubuntu-advantage-tools
Am I missing sometihng?

Reply:
Yeah, this didn't work in a test release-upgrade from precise, with the ppa enabled.
At the end, I had these files in `/etc/apt/`:
```
/etc/apt/preferences.d/ubuntu-esm-infra-trusty
/etc/apt/sources.list.d/ubuntu-esm-precise.list
/etc/apt/sources.list.d/ubuntu-esm-infra-trusty.list
```
`ubuntu-esm-precise.list` has these contents:
```
deb https://lol:secret@esm.ubuntu.com/ubuntu trusty-updates main
# deb-src https://lol:secret@esm.ubuntu.com/ubuntu precise main
deb https://lol:secret@esm.ubuntu.com/ubuntu trusty-security main
```
And `ubuntu-esm-infra-trusty.list`:
```
# Written by ubuntu-advantage-tools
deb https://esm.ubuntu.com/ubuntu trusty-infra-security main
# deb-src https://esm.ubuntu.com/ubuntu trusty-infra-security main

deb https://esm.ubuntu.com/ubuntu trusty-infra-updates main
# deb-src https://esm.ubuntu.com/ubuntu trusty-infra-updates main
```
The only reason we didn't get the apt warning this time is because the pockets have different names, trusty having the `infra` added term.
Furthermore, esm is disabled now, because of the pinning file in `/etc/apt/preferences.d/ubuntu-esm-infra-trusty`:
```
# Written by ubuntu-advantage-tools
Package: *
Pin: release o=UbuntuESM, n=trusty
Pin-Priority: never
```
`libcurl3`, for example, I have installed from the esm repository after this `do-release-upgrade`, but the pinning effectively disables it:
```
root@p1:~# apt-cache policy libcurl3
libcurl3:
  Installed: 7.35.0-1ubuntu2.20+esm3
  Candidate: 7.35.0-1ubuntu2.20+esm3
  Version table:
 *** 7.35.0-1ubuntu2.20+esm3 0
       -32768 https://esm.ubuntu.com/ubuntu/ trusty-infra-security/main amd64 Packages
       -32768 https://esm.ubuntu.com/ubuntu/ trusty-security/main amd64 Packages
        100 /var/lib/dpkg/status
     7.35.0-1ubuntu2.20 0
        500 http://archive.ubuntu.com/ubuntu/ trusty-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu/ trusty-security/main amd64 Packages
     7.35.0-1ubuntu2 0
        500 http://archive.ubuntu.com/ubuntu/ trusty/main amd64 Packages
root@p1:~# 
```
And if I try a `--reinstall`, it kind of fails, probably because it fetches the package from `trusty-infra-security` for which we have no credentials. The credentials we have are only for `trusty-security`, in the `ubuntu-esm-precise.list` (yes, precise) list file. Still, in the end, it downloaded something, maybe it tried both repositories (but only showed one):
```
root@p1:~# apt install --reinstall libcurl3
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 1 not upgraded.
Need to get 173 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 https://esm.ubuntu.com/ubuntu/ trusty-infra-security/main libcurl3 amd64 7.35.0-1ubuntu2.20+esm3 [173 kB]
Err https://esm.ubuntu.com/ubuntu/ trusty-infra-security/main libcurl3 amd64 7.35.0-1ubuntu2.20+esm3
  HttpError401
Fetched 173 kB in 3s (55.6 kB/s)
(Reading database ... 77767 files and directories currently installed.)
Preparing to unpack .../libcurl3_7.35.0-1ubuntu2.20+esm3_amd64.deb ...
Unpacking libcurl3:amd64 (7.35.0-1ubuntu2.20+esm3) over (7.35.0-1ubuntu2.20+esm3) ...
Setting up libcurl3:amd64 (7.35.0-1ubuntu2.20+esm3) ...
Processing triggers for libc-bin (2.19-0ubuntu6.15) ...
root@p1:~# echo $?
0
```

Reply:
This confirms the release upgrade upgraded ubuntu-advantage-tools version 1 from precise:
```
Unpacking python3-pkg-resources (3.3-1ubuntu2) ...
Preparing to unpack .../ubuntu-advantage-tools_19.6~642~git.986b779~ubuntu14.04.1_amd64.deb ...
Moving obsolete conffile /etc/update-motd.d/99-esm out of the way...
Unpacking ubuntu-advantage-tools (19.6~642~git.986b779~ubuntu14.04.1) over (1) ...
```